### PR TITLE
Create flow to update subgraph docs on Dev-Portal

### PR DESCRIPTION
**Pull request by banasa44**
## Description

This PR introduces everything needed to generate Subgraph documentation and send it to the Developer-Portal. It mimics the workflows on SDK and OSX/contracts.

It doesn't generate all the subraph documentation as the reference-guide documentation is generated in the Dev-Portal itself as we are using a plugin from Docusaurus that generates it. What we are providing though is the result of the introspection query needed to generate the subgraph docs.

The PR includes 2 query example samples as this will be the actual workflow, the test for these queries will be done in following-up tasks. 

Task ID: [OS-1051](https://aragonassociation.atlassian.net/browse/OS-1051)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-1051]: https://aragonassociation.atlassian.net/browse/OS-1051?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,4 +39,8 @@ managementDAOTX.json
 generated
 subgraph.yaml
 .bin
-mythx.json
\ No newline at end of file
+mythx.json
+
+# docs
+packages/subgraph/docs/developer-portal/01-query-examples/*
+!packages/subgraph/docs/developer-portal/01-query-examples/index.md
```
Comment (Rekard0) [inline .gitignore:46]:
```suggestion
!packages/subgraph/docs/developer-portal/01-query-examples/index.md

```

File: packages/subgraph/docs/codegen.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+generates:
+  ./schema-introspection.json:
+    schema: ../schema.graphql
+    plugins:
+      - introspection
```
Comment (Rekard0) [inline packages/subgraph/docs/codegen.yml:5]: can we move this file to the root/ && update the package.json

File: .gitignore (hunk 1 above)
Comment (Rekard0) [reply .gitignore:46]: i think we should also ignore `schema-introspection.json`

File: packages/subgraph/src/examples/01-dao-query.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          build
+          release {
+            release
+          }
+        }
+      }
+    }
+  }
+`;
+export const QueryDaos = gql`
```
Comment (Rekard0) [inline packages/subgraph/src/examples/01-dao-query.ts:37]:
```suggestion
`;

export const QueryDaos = gql`
```

File: packages/subgraph/package.json [hunk 4]
```diff
@@ -14,6 +14,8 @@
     "deploy": "scripts/deploy-subgraph.sh",
     "create:local": "graph create aragon/aragon-core-goerli --node http://localhost:8020",
     "deploy:local": "LOCAL=true scripts/deploy-subgraph.sh",
+    "generate:introspection": "cd docs && graphql-codegen && prettier schema-introspection.json --write",
```
Comment (Rekard0) [inline packages/subgraph/package.json:17]: once the .yml file is moved to root, this need to be updated

**Review by Michael-A-Heuer — CHANGES_REQUESTED** [review]
Let's make sure that the workflow does not run on closing a PR.
If this was found on the SDK, let's make sure to fix it there as well.

File: .github/workflows/subgraph-documentation-update.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# This workflow is triggered when a pull request is closed on the 'develop' or 'main' branches
+# and there is a change to the 'packages/subgraph/schema.graphql' file.
+
+name: Subgraph Documentation Update
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - closed
```
[the remark this answers is not in the document]
Comment (banasa44) [reply .github/workflows/subgraph-documentation-update.yml:10]: makes perfect sense, didn't thought about manually closed PRs

**Review by Michael-A-Heuer — APPROVED** [review]

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**